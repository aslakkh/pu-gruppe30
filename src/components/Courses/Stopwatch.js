/**
 * Created by anderssalvesen on 23.02.2017.
 */
import React, { Component } from 'react';
import './stopwatch.css';
import Progress from 'react-progressbar';
import {ref } from '../../config/constants'
import firebase from 'firebase';


const formattedSeconds = ((sec) => //formats to hh:mm:ss
Math.floor (sec/3600)+ ':' + Math.floor(sec / 60) + '.' + ('0' + sec % 60).slice(-2))


class Stopwatch extends Component {
    constructor(props) {
        super(props);
        this.state = {
            secondsElapsed: 0,
            lastClearedIncrementer: null,
            goal: 400,
            time: this.props.course.time
        };
        this.incrementer = null;
        this.started = false
    }

    castToFirebase(){
        const tid = this.state.time
    const userUid = firebase.auth().currentUser.uid;
    const timeRef = ref.child('users/'+userUid+'/courses/'+this.state.emne+'/sessions/'+ Date.now())
    timeRef.update({time:this.state.secondsElapsed+this.state.time})
    this.setState({
        secondsElapsed: 0,
        time: tid+this.state.secondsElapsed
    })
}
    handleStartClick() {
                if(!this.started){ //Makes sure the button isnt clicked twice
            this.started = true
        this.incrementer = setInterval( () =>
                this.setState({
                    secondsElapsed: this.state.secondsElapsed + 1
                })
            , 1000);}
    }

    handleStopClick() {
        this.started = false
        clearInterval(this.incrementer);
        this.setState({
            lastClearedIncrementer: this.incrementer
        });
    }




    componentWillMount(){
        this.setState ({
            secondsElapsed: 0,
            lastClearedIncrementer: null,
            goal: 400,
            emne: this.props.emne,
            time: this.props.course.time
        })
        this.incrementer = null;
        this.started = false

    }
    handleResetClick() {
        this.castToFirebase()
        clearInterval(this.incrementer);
    }

    render() {
        return (
            <div className="stopwatch">
                <h3>Total time spent:{formattedSeconds(this.state.secondsElapsed+this.state.time)}</h3>
                <h1 className="progressbar">
                    <Progress color={'#000000'} completed={(this.state.secondsElapsed >= this.state.goal) ? 100 : ((this.state.secondsElapsed+this.state.time)/this.state.goal) * 100}>
                    </Progress>
                </h1>
                <h1 className="app-timer">{formattedSeconds(this.state.secondsElapsed)}</h1>

                {(this.state.secondsElapsed === 0 ||
                    this.incrementer === this.state.lastClearedIncrementer
                        ? <Button className="start-btn" onClick={this.handleStartClick.bind(this)}>start</Button>
                        : <Button className="stop-btn" onClick={this.handleStopClick.bind(this)}>stop</Button>
                )}

                {(this.state.secondsElapsed !== 0 &&
                    this.incrementer === this.state.lastClearedIncrementer
                        ? <Button onClick={this.handleResetClick.bind(this)}>save</Button>
                        : null
                )}

            </div>

        );
    }
}

/** verbose component before 0.14
 class Button extends React.Component {
  render() {
    return <button type="button" {...this.props}
                   className={"btn " + this.props.className } />;
  }
}
 */

const Button = (props) =>
    <button type="button" {...props} className={"Stopwatch-btn " + props.className } />;

export default Stopwatch;

