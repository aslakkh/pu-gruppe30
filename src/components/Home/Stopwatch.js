import React, { Component } from 'react';
import Progress from 'react-progressbar';
import {ref } from '../../config/constants'
import firebase from 'firebase';
import {Button, ProgressBar, ListGroup, Grid,Col,Row} from 'react-bootstrap'
import PlannedSession from './PlannedSession'
import AddSession from './AddSession'

const formattedSeconds = ((sec) => //formats to hh:mm:ss
Math.floor (sec/3600)+ ':' + Math.floor(sec / 60) + '.' + ('0' + sec % 60).slice(-2));


class Stopwatch extends Component {
    constructor(props) {
        super(props);

        this.state = {
            secondsElapsed: 0,
            lastClearedIncrementer: null,
            goal: 400,
            desc: ""
        };
        this.incrementer = null;
        this.started = false
    }

    castToFirebase(){
        console.log("desc");
        console.log(this.state.desc.value);
    const userUid = firebase.auth().currentUser.uid;
    const timeRef = ref.child('users/'+userUid+'/courses/'+this.state.emne+'/sessions/'+ Date.now());
    timeRef.set({time:this.state.secondsElapsed,desc: this.state.desc.value});
    this.setState({
        secondsElapsed: 0
    })
}
    handleStartClick() {
        if(!this.started){ //Makes sure the button isnt clicked twice
            this.started = true;
            this.incrementer = setInterval( () =>
                this.setState({
                    secondsElapsed: this.state.secondsElapsed + 1
                })
            , 1000);}
    }

    handleStopClick() {
        this.started = false;
        clearInterval(this.incrementer);
        this.setState({
            lastClearedIncrementer: this.incrementer
        });
    }




    componentWillMount(){
        this.setState ({
            secondsElapsed: 0,
            lastClearedIncrementer: null,
            goal: 400,
            emne: this.props.emne,
            theme: undefined
        })
        this.incrementer = null;
        this.started = false
        console.log(this.props.emne)

    }
    handleResetClick() {
        this.castToFirebase();
        clearInterval(this.incrementer);
    }
    onChildChanged(newState){
        console.log(newState)
        this.setState({theme:newState})

    }

    render() {
        return (
            <Grid fluid={true}>
                <Row>
                    <Col md={7}>
            <div className="stopwatch" >
                <h4>Planned Session: {this.state.theme}</h4>
                <h1 className="app-timer">{formattedSeconds(this.state.secondsElapsed)}</h1>

                {(this.state.secondsElapsed === 0 ||
                    this.incrementer === this.state.lastClearedIncrementer
                        ? <Button block className="btn btn-lg" onClick={this.handleStartClick.bind(this)}>start</Button>
                        : <Button block className="btn btn-lg" onClick={this.handleStopClick.bind(this)}>stop</Button>
                )}

                {(this.state.secondsElapsed !== 0 &&
                    this.incrementer === this.state.lastClearedIncrementer
                        ? <div>  <form><label>What did you do?{this.state.desc}</label>
                            <input className="form-control" ref={(desc) => this.state.desc = desc} placeholder="Description"/>
                                </form>
                            <Button block className="btn btn-lg" onClick={this.handleResetClick.bind(this)}>save</Button></div>
                        : null
                )}

            </div><div>
                        <AddSession courseID = {this.props.emne}/>
                    </div>

                    </Col>
                    <Col md={1}/>
            <Col md={4}>

                <div>
                    <PlannedSession emne={this.props.emne} callbackParent={(newState) => this.onChildChanged(newState) }/>
                </div>
</Col>
                </Row>
            </Grid>
        );
    }
}


export default Stopwatch;

