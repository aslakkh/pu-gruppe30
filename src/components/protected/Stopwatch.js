/**
 * Created by anderssalvesen on 23.02.2017.
 */
import React, { Component } from 'react';
import './stopwatch.css';
import Progress from 'react-progressbar';


const formattedSeconds = (sec) =>
Math.floor(sec / 60) +
':' +
('0' + sec % 60).slice(-2)

function getFromFirebase(){

}

function castToFirebase(){

}

class Stopwatch extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            secondsElapsed: 0,
            lastClearedIncrementer: null,
            goal: 5
        };
        this.incrementer = null;
        this.started = false
    }

    handleStartClick() {
                if(!this.started){
            this.started = true
        this.incrementer = setInterval( () =>
                this.setState({
                    secondsElapsed: this.state.secondsElapsed + 1
                })
            , 1000);}
    }

    handleStopClick() {
                this.started = false
        clearInterval(this.incrementer);
        this.setState({
            lastClearedIncrementer: this.incrementer
        });
    }

    handleResetClick() {
        

        clearInterval(this.incrementer);
        let sec = this.state.secondsElapsed;

        this.setState({
            secondsElapsed: 0,
        });
    }

    render() {
        return (
            <div className="stopwatch">
                <h1 className="progressbar">
                    <Progress completed={(this.state.secondsElapsed >= this.state.goal) ? 100 : (this.state.secondsElapsed/this.state.goal) * 100}>
                    </Progress>
                </h1>
                <h1 className="app-timer">{formattedSeconds(this.state.secondsElapsed)}</h1>

                {(this.state.secondsElapsed === 0 ||
                    this.incrementer === this.state.lastClearedIncrementer
                        ? <Button className="start-btn" onClick={this.handleStartClick.bind(this)}>start</Button>
                        : <Button className="stop-btn" onClick={this.handleStopClick.bind(this)}>stop</Button>
                )}

                {(this.state.secondsElapsed !== 0 &&
                    this.incrementer === this.state.lastClearedIncrementer
                        ? <Button onClick={this.handleResetClick.bind(this)}>save</Button>
                        : null
                )}

            </div>

        );
    }
}

/** verbose component before 0.14
 class Button extends React.Component {
  render() {
    return <button type="button" {...this.props}
                   className={"btn " + this.props.className } />;
  }
}
 */

const Button = (props) =>
    <button type="button" {...props} className={"Stopwatch-btn " + props.className } />;

export default Stopwatch;

