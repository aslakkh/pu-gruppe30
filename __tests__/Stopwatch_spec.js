/**
 * Created by jan on 19/04/2017.
 */
import React from 'react';
import renderer from 'react-test-renderer';
import Stopwatch from '../src/components/Home/Stopwatch.js';
import TestUtils from "react-addons-test-utils"

const courses={
    TES1000:{active: true,
        goals: {dailyGoal:{active: false,
            timeSet: 0,
            timeSpent: 0,
            value: 0},
            monthlyGoal:
                {active:false,
                    timeSet:0,
                    timeSpent:0,
                    value:0},
            weeklyGoal:
                {active:false,
                    timeSet:0,
                    timeSpent:0,
                    value:0}

        }
    }};
const coursesNull=null;
describe("Stopwatch", function () {
    let renderer2;
    let component2;
    renderer2 = TestUtils.createRenderer();
    renderer2.render(
        <Stopwatch course={courses} emne={"TES1000"}/>
    );
    component2 = renderer2.getRenderOutput();
    it("One course", () => {
        expect(component2).toMatchSnapshot();
    });
});