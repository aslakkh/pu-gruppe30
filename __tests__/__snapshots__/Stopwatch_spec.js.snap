// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stopwatch One course 1`] = `
<Grid
  bsClass="container"
  componentClass="div"
  fluid={true}
>
  <Row
    bsClass="row"
    componentClass="div"
  >
    <Col
      bsClass="col"
      componentClass="div"
      md={7}
    >
      <div
        className="stopwatch"
      >
        <h4>
          Planned Session: 
        </h4>
        <h1
          className="app-timer"
        >
          0:0:00
        </h1>
        <Button
          active={false}
          block={true}
          bsClass="btn"
          bsStyle="default"
          className="btn btn-lg"
          disabled={false}
          onClick={[Function]}
        >
          start
        </Button>
      </div>
      <div>
        <AddSession
          courseID="TES1000"
        />
      </div>
    </Col>
    <Col
      bsClass="col"
      componentClass="div"
      md={1}
    />
    <Col
      bsClass="col"
      componentClass="div"
      md={4}
    >
      <div>
        <PlannedSession
          callbackParent={[Function]}
          emne="TES1000"
        />
      </div>
    </Col>
  </Row>
</Grid>
`;
